### PR TITLE
MAJOR FOUNDATIONS: Validations & Exception Handling

### PR DESCRIPTION
**Pull request by sunnynagavo**
Updated .gitignore with Visual Studio template.
Added Xeption package references to projects.
Implemented ILoggingBroker and LoggingBroker.
Enhanced EventAddressService with logging and validation. Added unit tests for validation and logging.

### CODE REVIEW

File: EventHighway.Core/Services/Foundations/EventAddresses/EventAddressService.Validations.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+    internal partial class EventAddressService
+    {
+        private void ValidateEventAddressOnAdd(EventAddress eventAddress)
+        {
+            ValidateEventAddressIsNotNull(eventAddress);
+        }
+
+        private void ValidateEventAddressIsNotNull(EventAddress eventAddress) =>
+            _ = eventAddress ?? throw new NullEventAddressException(message: "EventAddress is null");
```
Comment (glhays) [inline EventHighway.Core/Services/Foundations/EventAddresses/EventAddressService.Validations.cs:18]: reformat to make it readable by a level-0

File: EventHighway.Core/Services/Foundations/EventAddresses/EventAddressService.Exceptions.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                throw CreateAndLogValidationException(nullEventAddressException);
+            }
+        }
+        private EventAddressValidationException CreateAndLogValidationException(Xeption innerException)
+        {
+            var configurationValidationException =
+                new EventAddressValidationException(
+                    message: "EventAddress validation error occurred, fix errors and try again",
+                    innerException: innerException);
+            this.loggingBroker.LogErrorAsync(configurationValidationException);
```
Comment (glhays) [inline EventHighway.Core/Services/Foundations/EventAddresses/EventAddressService.Exceptions.cs:32]: line break here as well

File: EventHighway.Core/Clients/EventHighways/EventHighwayClient.cs [hunk 3]
```diff
@@ -61,6 +62,8 @@ private IServiceProvider RegisterServices()
                 .AddTransient<IStorageBroker, StorageBroker>(broker =>
                     new StorageBroker(this.dataConnectionString))
 
+                .AddTransient<ILoggingBroker, LoggingBroker>()
+
```
Comment (glhays) [inline EventHighway.Core/Clients/EventHighways/EventHighwayClient.cs:66]: remove this line break

**Comment by glhays** [comment]
@sunnynagavo The logging stuff should be in its own PR, And not really a MAJOR

File: EventHighway.Core.Tests.Unit/Services/Foundations/EventAddresses/EventAddressServiceTests.Logic.Add.cs [hunk 4]
```diff
@@ -47,5 +49,38 @@ await this.eventAddressService.AddEventAddressAsync(
 
             this.storageBrokerMock.VerifyNoOtherCalls();
         }
+
+        [Fact]
```
Comment (glhays) [inline EventHighway.Core.Tests.Unit/Services/Foundations/EventAddresses/EventAddressServiceTests.Logic.Add.cs:53]: This test should live in it's own partial class. "EventAddressServiceTests.Validations.Add"

**Comment by hassanhabib** [comment]
@sunnynagavo this PR has been open for a week now - we may need to close it due to inactivity.

**Comment by sunnynagavo** [comment]
We can close this @hassanhabib . I just reached back to seattle. let me send you another PR with proper commits.

**Comment by hassanhabib** [comment]
@sunnynagavo sounds good - and welcome back!

**Pull request closed without merging**